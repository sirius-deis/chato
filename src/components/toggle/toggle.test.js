import { render, screen, fireEvent } from "@testing-library/react";
import Toggle from "./toggle";

describe("Toggle component", () => {
  it("should match snapshot", () => {
    const { container } = render(<Toggle />);
    expect(container).toMatchSnapshot();
  });
  it("should render toggle with the right checked true value", () => {
    render(<Toggle checked />);
    const checkbox = screen.getByRole("checkbox");
    fireEvent.click(checkbox);
    expect(checkbox.checked).toEqual(true);
  });
  it("should render toggle with the right checked false value", () => {
    render(<Toggle checked={false} />);
    const checkbox = screen.getByRole("checkbox");
    fireEvent.click(checkbox);
    expect(checkbox.checked).toEqual(false);
  });
  it("should fire event", () => {
    const fn = jest.fn();
    render(<Toggle checked onChange={fn} />);
    fireEvent.click(screen.getByRole("checkbox"));
    expect(fn).toHaveBeenCalled();
  });
});
