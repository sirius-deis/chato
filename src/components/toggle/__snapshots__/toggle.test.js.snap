// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle component should match snapshot 1`] = `
<div>
  <label
    class="sc-aXZVg hzAiqk"
  >
    <input
      class="sc-gEvEer mMtQb"
      type="checkbox"
    />
    <span
      class="sc-eqUAAy kKOnaw"
    />
  </label>
</div>
`;
